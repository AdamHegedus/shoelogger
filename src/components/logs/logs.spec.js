import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import BootstrapVue from 'bootstrap-vue';
import Logs from '@/components/logs/logs';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueRouter);
localVue.use(BootstrapVue);

describe('logs.vue', () => {
    describe('when navigation is set', () => {
        it('should render component', () => {
            // ASSIGN
            const navigationState = {
                menu: {
                    logs: {
                        lastPage: '/logs/mock-navigation-state'
                    }
                }
            };

            const logsActions = {
                getLogs: jest.fn()
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    navigation: {
                        namespaced: true,
                        state: navigationState
                    },
                    logs: {
                        namespaced: true,
                        actions: logsActions
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions
                    }
                }
            });

            const router = new VueRouter();
            router.push = jest.fn();
            const wrapper = shallowMount(Logs, { store, localVue, router });

            // ACT
            const actual = wrapper.element;

            // ASSERT
            expect(actual).toMatchSnapshot();
            expect(logsActions.getLogs).toBeCalled();
            expect(shoesActions.getShoes).toBeCalled();
            expect(wrapper.vm.$router.push.mock.calls).toMatchSnapshot();
        });
    });

    describe('when navigation is unset', () => {
        it('should render component', () => {
            // ASSIGN
            const navigationState = {
                menu: {
                    logs: {
                        lastPage: null
                    }
                }
            };

            const logsActions = {
                getLogs: jest.fn()
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    navigation: {
                        namespaced: true,
                        state: navigationState
                    },
                    logs: {
                        namespaced: true,
                        actions: logsActions
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions
                    }
                }
            });

            const router = new VueRouter();
            router.push = jest.fn();
            const wrapper = shallowMount(Logs, { store, localVue, router });

            // ACT
            const actual = wrapper.element;

            // ASSERT
            expect(actual).toMatchSnapshot();
            expect(logsActions.getLogs).toBeCalled();
            expect(shoesActions.getShoes).toBeCalled();
            expect(wrapper.vm.$router.push.mock.calls).toMatchSnapshot();
        });
    });

    describe('newLog', () => {
        it('should call router', () => {
            // ASSIGN
            const navigationState = {
                menu: {
                    logs: {
                        lastPage: null
                    }
                }
            };

            const logsActions = {
                getLogs: jest.fn()
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    navigation: {
                        namespaced: true,
                        state: navigationState
                    },
                    logs: {
                        namespaced: true,
                        actions: logsActions
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions
                    }
                }
            });

            const router = new VueRouter();
            const wrapper = shallowMount(Logs, { store, localVue, router });
            const spy = jest.spyOn(wrapper.vm.$router, 'push');

            // ACT
            wrapper.find('button').trigger('click');

            // ASSERT
            expect(spy.mock.calls).toMatchSnapshot();
        });
    });
});
