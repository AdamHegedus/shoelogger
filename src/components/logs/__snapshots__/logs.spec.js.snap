// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logs.vue newLog should call router 1`] = `
Array [
  Array [
    "/logs/new",
  ],
]
`;

exports[`logs.vue when navigation is set should render component 1`] = `
<div
  class="container"
>
  <div
    class="float-right"
  >
    <button
      class="btn btn-primary"
    >
      
            New
        
    </button>
  </div>
   
  <h2>
    Logs
  </h2>
   
  <b-navbar-stub
    class="sub-navbar"
    tag="nav"
    type="light"
  >
    <b-navbar-nav-stub
      tag="ul"
    >
      <b-nav-item-stub
        activeclass="active"
        event="click"
        exactactiveclass="active"
        routertag="a"
        target="_self"
        to="/logs/overview"
      >
        
                Overview
            
      </b-nav-item-stub>
       
      <b-nav-item-stub
        activeclass="active"
        event="click"
        exactactiveclass="active"
        routertag="a"
        target="_self"
        to="/logs/recent"
      >
        
                Recent
            
      </b-nav-item-stub>
       
      <b-nav-item-stub
        activeclass="active"
        event="click"
        exactactiveclass="active"
        routertag="a"
        target="_self"
        to="/logs/by-shoes"
      >
        
                By Shoes
            
      </b-nav-item-stub>
    </b-navbar-nav-stub>
  </b-navbar-stub>
   
  <router-view-stub
    mode="out-in"
    name="slide"
  />
</div>
`;

exports[`logs.vue when navigation is set should render component 2`] = `
Array [
  Array [
    "/logs/mock-navigation-state",
  ],
]
`;

exports[`logs.vue when navigation is unset should render component 1`] = `
<div
  class="container"
>
  <div
    class="float-right"
  >
    <button
      class="btn btn-primary"
    >
      
            New
        
    </button>
  </div>
   
  <h2>
    Logs
  </h2>
   
  <b-navbar-stub
    class="sub-navbar"
    tag="nav"
    type="light"
  >
    <b-navbar-nav-stub
      tag="ul"
    >
      <b-nav-item-stub
        activeclass="active"
        event="click"
        exactactiveclass="active"
        routertag="a"
        target="_self"
        to="/logs/overview"
      >
        
                Overview
            
      </b-nav-item-stub>
       
      <b-nav-item-stub
        activeclass="active"
        event="click"
        exactactiveclass="active"
        routertag="a"
        target="_self"
        to="/logs/recent"
      >
        
                Recent
            
      </b-nav-item-stub>
       
      <b-nav-item-stub
        activeclass="active"
        event="click"
        exactactiveclass="active"
        routertag="a"
        target="_self"
        to="/logs/by-shoes"
      >
        
                By Shoes
            
      </b-nav-item-stub>
    </b-navbar-nav-stub>
  </b-navbar-stub>
   
  <router-view-stub
    mode="out-in"
    name="slide"
  />
</div>
`;

exports[`logs.vue when navigation is unset should render component 2`] = `
Array [
  Array [
    "/logs/overview",
  ],
]
`;
